src/dom/get-visible-y-from-root.ts

```diff
@@ -0,0 +1,47 @@
+import { dom } from '../';
+import { domVectorf } from './dom-vectorf';
+
+const ignoredPositions = new Set(['fixed', 'absolute']);
+
+function isFixed(element: HTMLElement) {
+  return dom.getStyle(element).position === 'fixed';
+}
+
+function getTransformedOffset(candidateElement: HTMLElement): number {
+  return candidateElement.offsetTop +
+      domVectorf.fromElementTransform(candidateElement).y;
+}
+
+function getVisibleDistanceFromRoot(
+    element: HTMLElement,
+    getOffsetFn: (e: HTMLElement) => number
+): number {
+  // Short circuit for fixed elements
```

Comment:
Small nit pick: Google style guides generally recommend comments that end in punctuation.

Reply:
Done.